llvm/lib/Transforms/IPO/PassManagerBuilder.cpp

```diff
@@ -1174,8 +1189,9 @@ void PassManagerBuilder::addLateLTOOptimizationPasses(
     PM.add(createHotColdSplittingPass());
 
   // Delete basic blocks, which optimization passes may have killed.
-  PM.add(
-      createCFGSimplificationPass(SimplifyCFGOptions().hoistCommonInsts(true)));
+  if (!SYCLOptimizationMode)
+    PM.add(
+        createCFGSimplificationPass(SimplifyCFGOptions().hoistCommonInsts(true)));
```

Comment:
It seems to me that the main thing we are fighting is hoisting common instructions - will it work if we simply pass an empty config here under `SYCLOptimizationMode` instead of completely disabling the pass everywhere?

Reply:
I've tried your guess. I think you are right. Left only this change.